### PR TITLE
docs(pubsub): add page describing retry policies

### LINKED ISSUE #11267: Document Pub/Sub retry and backoff policies

### PR DESCRIPTION
**Pull request by coryan**
Fixes #11267 Part of the work for #10157 

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11957)

### CODE REVIEW

File: google/cloud/pubsub/doc/override-retry-policies.dox [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+The policies can be set when the `*Connection` object is created. The library
+provides default policies for any policy that is not set. The application can
+also override some (or all) policies when the `*Client` object is created. This
+can be useful if multiple `*Client` objects share the same `*Connection` object,
+but you want different retry behavior in some of the clients. Finally, the
+application can override some retry policies when calling a specific member
+function.
+
+The library uses three different options to control the retry loop. The options
+have per-client names
```
Comment (devbww) [inline google/cloud/pubsub/doc/override-retry-policies.dox:23]: s/$/./

File: google/cloud/pubsub/doc/override-retry-policies.dox [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+(or is "truncated") if it reaches a prescribed maximum.
+
+@see google::cloud::pubsub::BackoffPolicyOption
+@see google::cloud::pubsub::SchemaServiceBackoffPolicyOption
+
+@section pubsub-override-retry-idempotency-policy Controlling which operations are retryable
+
+[`SchemaServiceClient`] supports the
+[`SchemaServiceConnectionIdempotencyPolicyOption`] to control which requests are
+retryable.  In all other cases the Pub/Sub library retries the operations.
```
Comment (devbww) [inline google/cloud/pubsub/doc/override-retry-policies.dox:69]: It is a little strange to not mention idempotency in an "idempotency policy".  Perhaps we should say something like "controls which operations are idempotent, and therefore retryable"?

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/doc/override-retry-policies.dox (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsub/doc/override-retry-policies.dox]: Done.

File: google/cloud/pubsub/doc/override-retry-policies.dox (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/doc/override-retry-policies.dox]: Done.

**Pull request merged**